### PR TITLE
723 team 02 project submission endpoints

### LINKED ISSUE #723: [TEAM-02] Project submission endpoints
## Overview
Save-in-progress and final-submit endpoints for team projects, time-gated.

## Goals
- Allow saving work in progress
- Allow final submission with validation
- Enforce the submission deadline

## Tasks
- [ ] `saveProject` — save while still working
- [ ] `submitProject` — final submit (lock after deadline)
- [ ] Anyone on the team can submit; validate all required details present
- [ ] Capture per-member DevPost + GitHub usernames
- [ ] Capture team DevPost URL + team GitHub URL
- [ ] Unit tests

## References
- Auth pattern (`protectedProcedure` + error handling): https://github.com/hackwestern/hackwestern/blob/dev/src/server/api/routers/scavenger-hunt.ts#L277-L304
- Unit-test pattern for edge cases: https://github.com/hackwestern/hackwestern/blob/dev/src/server/api/routers/application.test.ts#L21-L60

### PR DESCRIPTION
**Pull request by LucasVand**
closes (Github Issue number)

# Quick Overview of Changes

- added saveProject route which saves an in-progress project
- added submitProject route which validates a team and ensures that all fields are filled out and if so updates the time stamp and submission status, if submitted late it will still go through but be marked with late submission
- added simples tests for routes

# Checklist

- [ ] If any frontend changes were made, include a screenshot/demo in the overview
- [ ] Checked all uses of changed component(s)/function(s)
- [x] If any changes were made to our backend services, at least one (happy path) unit test was added OR a `TODO` comment was added to create one.
- [x] Check that CI is passing.
- [ ] If no (re-)reviews after 1-2 days, ping the web leads on the discord to remind them to review the PR.

@lucianlavric @JasmineGu2

### CODE REVIEW

File: src/server/api/routers/teams.test.ts [hunk 1]
```diff
@@ -150,4 +150,77 @@ describe("teams basic endpoints", () => {
       await expect(res).rejects.toThrow();
     });
   });
+  describe.sequential("submitProject Tests", () => {
+    test("submitProject fail", async () => {
```
Comment (lucianlavric) [inline src/server/api/routers/teams.test.ts]: Test names should be unique

**Review by lucianlavric — CHANGES_REQUESTED** [review]
fix duplicate submitProject fail tests

**Comment by lucianlavric** [comment]
I'll just fix the duplicate lol
